### PR TITLE
[RAPTOR-5561] Refine readme in DRUM repo stage 1

### PR DESCRIPTION
**Pull request by drenfr01**
# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

## Summary

Updating existing readme files and adding additional in child folders

## Rationale

The original README was too large. We are logically splitting it up into different files and removing content that is best served with a link instead.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,332 +1,72 @@
...
-        1) `GET /URL_PREFIX/stats/` This route is used to fetch memory usage data for DataRobot Custom Model Testing.
-        2) `GET /URL_PREFIX/health/` This route is used to check if model is loaded and functioning properly. If model loading fails error with 513 response code should be returned.  
-           Failing to handle this case may cause backend k8s container to enter crash/restart loop for several minutes.
-3) An executable `start_server.sh` file is required to start the model server.
-4) Any code and `start_server.sh` should be copied to `/opt/code/` by your Dockerfile
+* [Custom models walkthrough](https://community.datarobot.com/t5/knowledge-base/working-with-custom-models/ta-p/6082)
+* Code examples:
+    * [Custom inference models templates](https://github.com/datarobot/datarobot-user-models/tree/master/model_templates)
+    * [Environment Templates](https://github.com/datarobot/datarobot-user-models/tree/master/public_dropin_environments)
+* How to define a custom inference model:
```
Comment (MikhailYakubovskiy) [inline README.md:54]: shall we call it "A reference for defining a custom inference model:" ?

File: README.md [hunk 2]
```diff
@@ -1,332 +1,72 @@
...
+    * [Assembling a code folder](https://docs.google.com/document/d/154C_dDDZJlBClK4DjCOXQJlos6jszicgOmcCX14Mm_s/edit#heading=h.q52wbalbzdy9)
```
Comment (MikhailYakubovskiy) [inline README.md:55]: Let's make it actionable, in line with other names: "Assemble a code folder"

File: README.md [hunk 3]
```diff
@@ -1,332 +1,72 @@
...
+    * [Define a structured model](https://docs.google.com/document/d/154C_dDDZJlBClK4DjCOXQJlos6jszicgOmcCX14Mm_s/edit#heading=h.rd7ksjij0hhn)
+    * [Define an unstructured model](https://docs.google.com/document/d/154C_dDDZJlBClK4DjCOXQJlos6jszicgOmcCX14Mm_s/edit#heading=h.wg9ykdry860i)
```
Comment (MikhailYakubovskiy) [inline README.md:57]: We'll need to replace all the links that point to google docs with links that point to the respective sections on github

File: README.md [hunk 4]
```diff
@@ -1,332 +1,72 @@
...
+    * [Test a model with DRUM locally](https://docs.google.com/document/d/154C_dDDZJlBClK4DjCOXQJlos6jszicgOmcCX14Mm_s/edit#heading=h.wx9siz7rht7w)
+    * [Build your own environment](https://docs.google.com/document/d/154C_dDDZJlBClK4DjCOXQJlos6jszicgOmcCX14Mm_s/edit#heading=h.d59dwsyskndu)
+    * [Upload, manage, and deploy a model in DataRobot](https://app.datarobot.com/docs/mlops/deployment/custom-models/custom-inf-model.html)
 
+Other sources:
+* There is a chance that the task you are looking for has already been implemented.
```
Comment (MikhailYakubovskiy) [inline README.md:63]: Let's replace "task" with "model" in here

File: model_templates/README.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+* [PyTorch sample model](model_templates/python3_pytorch)
+* [XGBoost sample model](model_templates/python3_xgboost)
+* [Keras sample model](model_templates/python3_keras)
+* [Keras sample model + Joblib artifact](model_templates/python3_keras_joblib)
+* [PyPMML sample model](model_templates/python3_pmml)
+* [R sample model](model_templates/r_lang)
+* [Java sample model](model_templates/java_codegen)
+* [Julia sample models](model_templates/julia)
+
+##### Training Models
```
Comment (MikhailYakubovskiy) [inline model_templates/README.md:22]: Let's remove the training models section from here.

File: public_dropin_environments/README.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
+ # Custom Environment Templates
+The [public_dropin_environments](https://github.com/datarobot/datarobot-user-models/tree/master/public_dropin_environments)
+contains templates for the base environments used in DataRobot.
+Dependency requirements can be applied to the base environment to create a
+runtime environment for both custom tasks and/or custom inference models.
+A custom environment defines the runtime environment for either a custom task 
+or custom inference model.
+In this repository, we provide several example environments that you can use and modify:
```
Comment (MikhailYakubovskiy) [inline public_dropin_environments/README.md:8]: let's fix all the links to environments

File: public_dropin_environments/README.md [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+* [Python 3 + PyTorch](public_dropin_environments/python3_pytorch)
+* [Python 3 + xgboost](public_dropin_environments/python3_xgboost)
+* [Python 3 + keras/tensorflow](public_dropin_environments/python3_keras)
+* [Python 3 + pmml](public_dropin_environments/python3_pmml)
+* [R + caret](public_dropin_environments/r_lang)
+* [Java Scoring Code](public_dropin_environments/java_codegen)
+* [Julia + MLJ](public_dropin_environments/julia_mlj)
+
+These sample environments each define the libraries available in the environment 
+and are designed to allow for simple custom tasks and/or custom inference models to be made that
```
Comment (MikhailYakubovskiy) [inline public_dropin_environments/README.md:19]: here the statement refers to custom inference models. Let's either not mention custom tasks here or refine it to make it more relevant for custom tasks.

File: DEFINE-INFERENCE-MODEL.md [hunk 8]
```diff
@@ -0,0 +1,283 @@
...
+Note: See overview at [README](https://github.com/datarobot/datarobot-user-models#readme)
+
+## Content
+
+1. [Assembling an inference model code folder](#inference_model_folder)
+2. [Define structured inference model](#structured_inference_model)
+3. [Define unstructured inference model](#unstructured_inference_model)
+4. [Test an inference model with DRUM locally](#test_inference_model_drum)
+5. [Building your own environment](#build_own_environment)
+6. [Upload, manage, and deploy a custom inference model with DataRobot](#custom_model_runner)
```
Comment (MikhailYakubovskiy) [inline DEFINE-INFERENCE-MODEL.md:12]: should we add a link to DR docs here? we actually have a page that describes how to upload, manage, etc custom models: https://app.datarobot.com/docs/mlops/deployment/custom-models/custom-inf-model.html

File: DEFINE-INFERENCE-MODEL.md [hunk 9]
```diff
@@ -0,0 +1,283 @@
+# How to define a custom inference model 
+
+Note: See overview at [README](https://github.com/datarobot/datarobot-user-models#readme)
+
+## Content
+
+1. [Assembling an inference model code folder](#inference_model_folder)
+2. [Define structured inference model](#structured_inference_model)
+3. [Define unstructured inference model](#unstructured_inference_model)
+4. [Test an inference model with DRUM locally](#test_inference_model_drum)
```
Comment (MikhailYakubovskiy) [inline DEFINE-INFERENCE-MODEL.md:10]: the link is broken here

File: DEFINE-INFERENCE-MODEL.md [hunk 10]
```diff
@@ -0,0 +1,283 @@
...
+custom environment:
+
+- a serialized model artifact with a file extension corresponding to the chosen environment language.
+- any additional custom code required to use it.
+
+The `drum new model` command can help you generate a model template.
+Check [here](https://github.com/datarobot/datarobot-user-models/tree/master/custom_model_runner#model-template-generation) for more information.
+
+See [here](https://github.com/datarobot/datarobot-user-models/tree/master/model_templates) 
+for examples you can just as a template.
```
Comment (MikhailYakubovskiy) [inline DEFINE-INFERENCE-MODEL.md:34]: "for examples you can use as a template."

**Review by sephmard — APPROVED** [review]

File: DEFINE-INFERENCE-MODEL.md (hunk 9 above)
Comment (drenfr01) [reply DEFINE-INFERENCE-MODEL.md:10]: fixed

File: DEFINE-INFERENCE-MODEL.md (hunk 8 above)
Comment (drenfr01) [reply DEFINE-INFERENCE-MODEL.md:12]: agreed! Added

File: README.md (hunk 1 above)
Comment (drenfr01) [reply README.md:54]: I like it! Done

File: README.md (hunk 2 above)
Comment (drenfr01) [reply README.md:55]: done

File: README.md (hunk 4 above)
Comment (drenfr01) [reply README.md:63]: done

File: model_templates/README.md (hunk 5 above)
Comment (drenfr01) [reply model_templates/README.md:22]: done

File: public_dropin_environments/README.md (hunk 6 above)
Comment (drenfr01) [reply public_dropin_environments/README.md:8]: fixed

File: public_dropin_environments/README.md (hunk 7 above)
Comment (drenfr01) [reply public_dropin_environments/README.md:19]: agreed. I just removed the reference to custom tasks for now and we can clean it up further in stage 2 once all the custom tasks documents are ready

File: model_templates/README.md [hunk 11]
```diff
@@ -0,0 +1,32 @@
+# Custom Inference Model Templates
```
[the remark this answers is not in the document]
Comment (drenfr01) [reply model_templates/README.md:1]: done

File: README.md (hunk 3 above)
Comment (drenfr01) [reply README.md]: done!

**Review by MikhailYakubovskiy — APPROVED** [review]
Looks great!

**Review by scottp-ml — APPROVED** [review]

**Pull request merged**